wear/src/main/java/io/homeassistant/companion/android/HomeAssistantApplication.kt

```diff
@@ -11,14 +11,32 @@ import android.nfc.NfcAdapter
 import android.os.Build
 import android.os.PowerManager
 import dagger.hilt.android.HiltAndroidApp
+import io.homeassistant.companion.android.common.data.keychain.KeyChainRepository
+import io.homeassistant.companion.android.common.data.keychain.KeyStoreImpl
 import io.homeassistant.companion.android.complications.ComplicationReceiver
 import io.homeassistant.companion.android.sensors.SensorReceiver
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Named
 
 @HiltAndroidApp
 open class HomeAssistantApplication : Application() {
+    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + Job())
```

Comment:
When creating a long-lived `CoroutineScope`, it's better to use `SupervisorJob`, as the scope won't get canceled if one coroutine fails. Now `ioScope` is only used at a single place, but in the future, others can also start using it.

```suggestion
    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + SupervisorJob())
```

Reply:
Not sure if I understood this correctly, `CoroutineScope(Dispatchers.IO + Job())` is already used 7x in "common" and "app" modules in the source code. The new addition is this line in the "wear" module. Should all other occurrences be changed to `SupervisorJob()` as well?

Reply:
@marazmarci While [you're not wrong](https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad#:~:text=You%20can%20call%20it%20applicationScope%20and%20it%20must%20contain%20a%20SupervisorJob()%20so%20that%20failures%20in%20coroutines%20don%E2%80%99t%20propagate%20in%20the%20hierarchy%20(as%20seen%20in%20part%203%20of%20the%20series)%3A) let's not change a common pattern in this PR, that's more (unrelated) changes. Feel free to make that change in another PR. Using a 'normal' job works if you're aware of it and handle exceptions.